ops/jenkins/global-pipeline-libraries/vars/awsAuth.groovy

```diff
@@ -20,11 +20,11 @@
  */
 void assumeRole(Map args = [:]) {
 	sessionDurationSeconds = args.sessionDurationSeconds ?: 3600
-	// default `sessionName` as `env.JOB_NAME` is vaguely sanitized to replace non-alphanumeric
+	// default `sessionName` as `env.JOB_BASE_NAME` is vaguely sanitized to replace non-alphanumeric
 	// chars with '-' AND deduplicates consecutive '-'. This is somewhat more restrictive than the
 	// regex used in the STS API for RoleSessionName, i.e. `/[\w+=,.@:\/-]*/`
 	// See https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html
-	sessionName = args.sessionName ?: env.JOB_NAME.replaceAll(/[^a-zA-Z\d]/, '-').replaceAll(/[\-]+/, '-')
+	sessionName = args.sessionName ?: env.JOB_BASE_NAME.replaceAll(/[^a-zA-Z\d]/, '-').replaceAll(/[\-]+/, '-')
```

Comment:
`env.JOB_BASE_NAME` might be more appropriate. What do these session names look like in practice? Does this avoid issues that might result from arbitrarily long branch names, too?